src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -407,6 +408,11 @@ private static string GetAssemblyQualifiedName(Type type)
                         tasks.Add(AnalyzeDocumentAsync(suppressionAnalyzer, document, span: null, bag.Add));
                     }
 
+                    foreach (var document in await project.GetSourceGeneratedDocumentsAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        tasks.Add(AnalyzeDocumentAsync(suppressionAnalyzer, document, span: null, bag.Add));
```

Comment:
Why do we want to show unnecessary suppressions on a source generated document? Is it possible to apply the remove unnecessary suppression code fix or would the fading just be a hint to update the source generator to stop generating the unnecessary pragma? Given that the source generator might be enabled on different projects with different set of analyzer references, I am not sure we should run this analyzer on source generated documents.